src/components/transcript/transcript.tsx

```diff
@@ -110,12 +111,22 @@ export class Transcript extends Component<TranscriptProps, TranscriptState> {
   private _bottomAutoscrollEdge: number = 0;
   private _thirdOfWidgetHeight: number = 0;
 
+  private _resizeObserver: ResizeObserver | null = null;
+
   state: TranscriptState = {
     isAutoScrollEnabled: true,
     widgetWidth: 0,
     ...initialSearch
   };
 
+  componentDidMount(): void {
+    // observe transcript root element size changes
+    this._resizeObserver = new ResizeObserver(() => {
```

Comment:
i see that we have checks around ResizeObserver like here
https://github.com/kaltura/playkit-js/blob/d631460bf276db9d6e8ea5ee4961bedf5f9c93a1/src/utils/resize-watcher.ts#L36

Need to see if we need this logic here as well. Using window.ResizeObserver sounds good, not sure about whether we need iframe observer though.

Reply:
use redux instead of ResizeObserver